src/codewhisperer/util/commonUtil.ts

```diff
@@ -27,3 +32,43 @@ export function isAwsError(error: any): boolean {
         error.time instanceof Date
     )
 }
+
+export function getFileExt(languageId: string) {
+    switch (languageId) {
+        case 'java':
+            return '.java'
+        case 'python':
+            return '.py'
+        default:
+            break
+    }
+    return undefined
+}
+
+export async function getDirSize(
+    dirPath: string,
+    startTime: number,
+    duration: number,
+    fileExt: string
+): Promise<number> {
+    if (performance.now() - startTime > duration) return 0
+    const files = await fs.readdir(dirPath, { withFileTypes: true })
+    const fileSizes = files.map(async file => {
+        const filePath = path.join(dirPath, file.name)
+        if (file.isDirectory()) return getDirSize(filePath, startTime, duration, fileExt)
+        if (file.isFile() && file.name.endsWith(fileExt)) {
+            const { size } = await fs.stat(filePath)
+            return size
+        }
+        return 0
+    })
+    return (await Promise.all(fileSizes)).reduce((accumulator, size) => accumulator + size, 0)
+}
+
+export function getWorkspaceFolder(uri: vscode.Uri) {
+    const workspaceFolder = vscode.workspace.getWorkspaceFolder(uri)
+    if (workspaceFolder === undefined) {
+        return path.dirname(uri.fsPath)
```

Comment:
If the `getWorkspaceFolder` call fails, this returns a non-workspace function. So the `getWorkspaceFolder` name is misleading. 

Also the purpose of this function isn't obvious. Needs a short (less than 10 words) docstring.

Reply:
This is a good point, I will need to change the logic little bit; for no workspaceFolder use case, it usually means customer is working on a single file, in this case, I should prevent doing directory traverse since the directory contains the file is not working project.